### PR TITLE
fix(gcp-functions): Change Node-API logic to use preview feature in schema instead of env var for this project (because env var not available during installation / post install script)

### PR DESCRIPTION
**Pull request by janpio**
Accidentally committed the improved shell script to the main branch of Will, but at least the exclusion I only removed here 🪀 

Then I fixed the problem.

Part of https://github.com/prisma/e2e-tests/issues/1748

### CODE REVIEW

**Comment by janpio** [comment]
Here is more log output via https://console.cloud.google.com/functions/details/us-central1/e2e-test-2021-06-09-194410?project=prisma-e2e-tests-265911&tab=logs

```
2021-06-09T19:45:39.809Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.809Z prisma:tryLoadEnv Environment variables not found at null
2021-06-09T19:45:39.809Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.809Z prisma:tryLoadEnv Environment variables not found at undefined
2021-06-09T19:45:39.809Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.809Z prisma:tryLoadEnv No Environment variables loaded
2021-06-09T19:45:39.812Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.811Z prisma:client clientVersion: 2.25.0-dev.26
2021-06-09T19:45:39.827Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.827Z prisma:client:napi internalSetup
2021-06-09T19:45:39.991Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.991Z prisma:client:napi Search for Query Engine Library in /workspace/node_modules/.prisma/client
2021-06-09T19:45:39.992Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.992Z prisma:client:napi Search for Query Engine Library in /workspace/node_modules/@prisma/client
2021-06-09T19:45:39.992Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.992Z prisma:client:napi Search for Query Engine Library in /workspace/node_modules/@prisma/client
2021-06-09T19:45:39.992Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.992Z prisma:client:napi Search for Query Engine Library in /workspace/node_modules/.prisma/client
2021-06-09T19:45:39.992Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.992Z prisma:client:napi Search for Query Engine Library in /workspace/prisma
2021-06-09T19:45:39.993Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.992Z prisma:client:napi Search for Query Engine Library in /tmp/prisma-engines
2021-06-09T19:45:39.993Ze2e-test-2021-06-09-194410 2021-06-09T19:45:39.993Z prisma:client:napi Search for Query Engine Library in /workspace/node_modules/.prisma/client
2021-06-09T19:45:39.995Ze2e-test-2021-06-09-194410 Unhandled rejection
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 Error: Query engine library for current platform "debian-openssl-1.1.x" could not be found.
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 You incorrectly pinned it to debian-openssl-1.1.x
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 This probably happens, because you built Prisma Client on a different platform.
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 (Prisma Client looked in "/workspace/node_modules/@prisma/client/runtime/libquery_engine_napi-debian-openssl-1.1.x.so.node")
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 Searched Locations:
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 /workspace/node_modules/.prisma/client
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 /workspace/node_modules/@prisma/client
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 /workspace/node_modules/@prisma/client
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 /workspace/node_modules/.prisma/client
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 /workspace/prisma
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 /tmp/prisma-engines
2021-06-09T19:45:39.998Ze2e-test-2021-06-09-194410 /workspace/node_modules/.prisma/client
2021-06-09T19:45:39.999Ze2e-test-2021-06-09-194410 You already added the platforms "native", "debian-openssl-1.1.x" to the "generator" block
2021-06-09T19:45:39.999Ze2e-test-2021-06-09-194410 in the "schema.prisma" file as described in https://pris.ly/d/client-generator,
2021-06-09T19:45:39.999Ze2e-test-2021-06-09-194410 but something went wrong. That's suboptimal.
2021-06-09T19:45:39.999Ze2e-test-2021-06-09-194410 Please create an issue at https://github.com/prisma/prisma/issues/new
2021-06-09T19:45:39.999Ze2e-test-2021-06-09-194410 at NAPIEngine.getLibQueryEnginePath (/workspace/node_modules/@prisma/client/runtime/index.js:22968:15)
2021-06-09T19:45:40.052824ZCloud FunctionsCreateFunctionus-central1:e2e-test-2021-06-09-194410test-930@prisma-e2e-tests-265911.iam.gserviceaccount.com {@type: type.googleapis.com/google.cloud.audit.AuditLog, authenticationInfo: {…}, methodName: google.cloud.functions.v1.CloudFunctionsService.CreateFunction, resourceName: projects/prisma-e2e-tests-265911/locations/us-central1/functions/e2e-test-2021-06-09-194410, serviceName: cloudfunctions.googlea…
2021-06-09T19:45:41.014Ze2e-test-2021-06-09-194410 Error: Process exited with code 16 at process.on.code (/layers/google.nodejs.functions-framework/functions-framework/node_modules/@google-cloud/functions-framework/build/src/invoker.js:275:22) at process.emit (events.js:198:13) at process.EventEmitter.emit (domain.js:448:20) at process.exit (internal/process/per_thread.js:168:15) at Object.sendCrashResponse (/layers/google.nodejs.functions-framework/functions-framework/node_modules/@google-cloud/functions-framework/build/src/logger.js:37:9) at process.on.err (/layers/google.nodejs.functions-framework/functions-framework/node_modules/@google-cloud/functions-framework/build/src/invoker.js:271:22) at process.emit (events.js:198:13) at process.EventEmitter.emit (domain.js:448:20) at emitPromiseRejectionWarnings (internal/process/promises.js:140:18) at process._tickCallback (internal/process/next_tick.js:69:34)
2021-06-09T19:45:41.352801151Ze2e-test-2021-06-09-194410icpvzn0gp3g2 Function execution started
2021-06-09T19:45:42.064075967Ze2e-test-2021-06-09-194410icpvzn0gp3g2 Error: function terminated. Recommended action: inspect logs for termination reason. Additional troubleshooting documentation can be found at https://cloud.google.com/functions/docs/troubleshooting#logging Function invocation was interrupted.
Error: function terminated. Recommended action: inspect logs for termination reason. Additional troubleshooting documentation can be found at https://cloud.google.com/functions/docs/troubleshooting#logging Function invocation was interrupted.
2021-06-09T19:45:42.464864895ZError detected in e2e-test-2021-06-09-194410

**Comment by janpio** [comment]
As the previous comment showed that the log showed that it was not finding the expected engine file at any of the expected locations and so Prisma Client could not be initialized at all but crashed, I changed it to not do any Prisma stuff and just output the `files` in the `.prisma/client` location:

``` 
files":["index-browser.js","index.d.ts","index.js","package.json","query-engine-debian-openssl-1.1.x","schema.prisma"]
```

So the _binary_ is present only instead of `libquery_engine_napi-debian-openssl-1.1.x.so.node`.

Theory: The project uses `binaryTargets`, maybe that is causing problems: https://github.com/prisma/e2e-tests/blob/dev/platforms-serverless/gcp-functions/prisma/schema.prisma#L3

**Comment by janpio** [comment]
Removing the `binaryTargets` does not change that the wrong file is present in the Napi version of things.

**Comment by janpio** [comment]
It seems currently the `node_modules` directly is completely ignored for the upload and everything is skipped: https://github.com/prisma/e2e-tests/blob/janpio/check-binaries-gcp/platforms-serverless/gcp-functions/.gcloudignore#L15 The log of that skipping includes the correct engine files when napi is enabled:

```
DEBUG: Skipping file [./node_modules/.prisma/client/libquery_engine_napi-debian-openssl-1.1.x.so.node]
```

Where does the engine on the deployment even come from?

**Comment by janpio** [comment]
Seems there is an `npm install --production` step after the function is uploaded: https://cloud.google.com/functions/docs/writing/specifying-dependencies-nodejs#configuring_deployment_dependencies Possibly the env var is not set when that is executed. (That could mean that setting the preview feature in the schema could solve the problem)

**Comment by janpio** [comment]
Yep, now it works - so the ENV var is just not available here during the dependency installation process.

File: platforms-serverless/gcp-functions/prisma/schema.prisma [hunk 1]
```diff
@@ -1,6 +1,6 @@
 generator client {
   provider      = "prisma-client-js"
-  binaryTargets = ["native", "debian-openssl-1.1.x"]
```
Comment (janpio) [inline platforms-serverless/gcp-functions/prisma/schema.prisma:3]: Seems this was not even needed any more for GCP Functions.

**Comment by janpio** [comment]
(The failing test is unrelated and can be ignored to merge)

File: platforms-serverless/gcp-functions/run.sh [hunk 2]
```diff
@@ -5,6 +5,16 @@ set -eux
 # just make sure this variable is set because gcloud expects it to be set
 echo "$GCP_FUNCTIONS_ACCOUNT"
 
+# When PRISMA_FORCE_NAPI is set, overwrite existing schema file with one that enables the napi preview feature
+if [[ -z "${PRISMA_FORCE_NAPI+x}" ]]; then
+  # use the default schema at prisma/schema.prisma file
+  true
+else
+  ls -l
+  ls -l prisma
```
Comment (janpio) [inline platforms-serverless/gcp-functions/run.sh]:
```suggestion
```

**Review by williamluke4 — APPROVED** [review]

**Pull request merged**